app/code/community/Razorpay/Payments/Helper/Data.php

```diff
@@ -43,22 +87,32 @@ public function createOrder($order)
 
         $currency = Razorpay_Payments_Model_Paymentmethod::CURRENCY;
 
-        $url = $this->getRelativeUrl('order');
+        $razorpayOrderId = Mage::getSingleton('core/session')->getRazorpayOrderID();
 
-        $data = array(
-            'receipt'         => $orderId,
-            'amount'          => $amount,
-            'currency'        => $currency,
-            'payment_capture' => 1
-        );
+        if ($razorpayOrderId === null)
+        {
+            Mage::log(['razorpayOrderId' => 'NULL']);
```

Comment:
@captn3m0 It won't break, it just won't log a null order id. I initially considered logging the else case, but your call. However, I would like some logging to be added to our magento from here on - it will make trouble shooting a lot quicker in the future.